app/src/main/java/com/github/featuredetectandroid/MainActivity.kt

```diff
@@ -1,27 +1,138 @@
 package com.github.featuredetectandroid
 
+import android.Manifest
+import android.content.pm.PackageManager
 import android.os.Bundle
+import android.util.Log
+import android.widget.Toast
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.activity.viewModels
+import androidx.camera.core.CameraSelector
+import androidx.camera.core.ImageAnalysis
+import androidx.camera.core.ImageAnalysis.OUTPUT_IMAGE_FORMAT_YUV_420_888
+import androidx.camera.lifecycle.ProcessCameraProvider
 import androidx.compose.foundation.layout.Box
 import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.material.FloatingActionButton
-import androidx.compose.ui.Alignment
+import androidx.compose.material.Text
 import androidx.compose.ui.Modifier
+import androidx.core.content.ContextCompat
+import com.github.featuredetectandroid.ui.GrayscaleView
+import com.github.featuredetectandroid.ui.GrayscaleViewModel
 import com.github.featuredetectandroid.ui.theme.Theme
+import com.github.featuredetectandroid.utils.PhotoAnalyzer
+import java.util.concurrent.Executors
+
+private const val TAG = "MainActivity"
 
 class MainActivity : ComponentActivity() {
+    private val cameraExecutor = Executors.newSingleThreadExecutor()
+    private val imageViewModel by viewModels<GrayscaleViewModel>()
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+
+        tryStartCamera()
         setContent {
             Theme {
                 Box(Modifier.fillMaxSize()) {
-                    FloatingActionButton(
-                        onClick = { /*TODO*/ },
-                        modifier = Modifier.align(Alignment.BottomEnd)
-                    ) { }
+                    if (ContextCompat.checkSelfPermission(
+                            this@MainActivity,
+                            Manifest.permission.CAMERA
+                        ) == PackageManager.PERMISSION_GRANTED
+                    ) {
+                        GrayscaleView(
+                            imageViewModel.grayscaleBitmap,
+                            imageViewModel.width,
+                            imageViewModel.height
+                        )
+                    } else {
+                        Text(text = "Camera permission required")
```

Comment:
This text appears in the corner of the screen. You need to configure center alignment in the `Box`.

Reply:
fixed